hwcontest.md

```diff
@@ -10,59 +10,78 @@ active_tab: homework
-    * Your source code from the `answer` directory as a zip file `source.zip` produced by running `python3 zipsrc.py` must be uploaded to the `Final Project` submission page on [Coursys]({{ site.coursys }}).
-    * Your output on the testcases which is the file `output.zip` produced by running `python3 zipout.py` must be uploaded to the `Final Project` submission page on [Coursys]({{ site.coursys }}). When we run `check.py` on the public testcases it should have a value higher than the output from the default program to get any marks.
-    * Your testcases for the Compiler Contest which is the file `contest.zip` produced by running `python3 zipcontest.py` must be uploaded to the `Final Project` submission page on [Coursys]({{ site.coursys }}). When we run `check.py` on the public testcases it should have a value higher than the output from the default program to get any marks.
-* Your source code from `source.zip` must be on your gitlab repository.
+* For Part 1, you must turn in one thing:
+    * Your testcases for the Compiler Contest (the file `contest.zip` produced by running `python3 zipcontest.py`) must be uploaded to the `Project testcases` submission page on [Coursys]({{ site.coursys }}).
+* For Part 2, you must turn in two things:
+    * Your source code from the `answer` directory as a zip file `source.zip` (produced by running `python3 zipsrc.py`) must be uploaded to the `Final Project` submission page on [Coursys]({{ site.coursys }}).
+    * Your output on the contest testcases (which is the file `output.zip` produced by running `python3 zipout.py` after unzipping the new cases) must be uploaded to the `Final Project` submission page on [Coursys]({{ site.coursys }}). 
+* Your source code from `source.zip` must also be pushed to your gitlab repository.
 * Make sure that we can run `make decafcomp` in your answer directory to create the `decafcomp` binary.
 * You cannot use data or code resources outside of what is provided to you. If you use external code snippets provide citations in the `answer/README.md` file.
 * For the written description of your submission and supporting documentation, you can use plain ASCII but for math equations it is better to use kramdown. Do not use any proprietary or binary file formats such as Microsoft Word.
 
 ### Grading
 
-1. The compiler from each participating submission will be run on all the collected testcases. 
-1. Your compiler will be graded based on how many testcases were successfully passed and how many of your testcases were failed by other compilers in the contest.
-
+1. In Part 1, you will earn 1 point for each testcase which passes validation, to a maximum of 10 points.
+1. In Part 2, you will earn 1 point for each testcase on which your compiler produces the correct output. You will earn 1 bonus point each time one of your testcases causes another student's compiler to crash or produce the wrong output. The maximum number of bonus points you can earn is 10% of the overall grade.
```

Comment:
I updated the scoring so that harder cases earn more points. I think there should be a cap on the number of bonus points (otherwise someone could get full marks on part 2 by just submitting one really hard test case) but I'm happy to change that cap from 10%.